nixos/modules/installer/tools/nixos-install.sh

```diff
@@ -132,7 +138,9 @@ else
     system_root=$closure
     # Create a temporary file ending in .closure (so nixos-prepare-root knows to --import it) to transport the store closure
     # to the filesytem we're preparing. Also delete it on exit!
-    nix-store --export $(nix-store -qR $closure) > $system_closure
+    # Run in background to avoid blocking while trying to write to the FIFO
+    # $system_closure refers to
+    nix-store --export $(nix-store -qR $closure) > $system_closure &
```

Comment:
We already have concurrency of the same form going on (just expressed with a different syntax) everywhere that pipes are used ;) but yes, if `nix copy` allows doing this without any of that I absolutely agree that it's a better idea.